core-web/libs/portlets/edit-ema/portlet/src/lib/edit-ema-editor/components/ema-page-dropzone/ema-page-dropzone.component.ts

```diff
@@ -36,114 +31,51 @@ const POINTER_INITIAL_POSITION = {
-
-        const data: ClientData = JSON.parse(target.dataset.payload);
-        const isTop = this.isTop(event);
-
-        const insertPosition = isTop ? 'before' : 'after';
-
-        const payload = <PositionPayload>{
-            ...data,
-            position: insertPosition
-        };
-
-        this.place.emit(payload);
-        this.pointerPosition = POINTER_INITIAL_POSITION;
-    }
-
-    /**
-     * Emit place event and reset pointer position
+     * Set the pointer position
      *
      * @param {DragEvent} event
      * @memberof EmaPageDropzoneComponent
      */
-    onDropEmptyContainer(event: DragEvent): void {
+    onDragover(event: DragEvent): void {
         const target = event.target as HTMLDivElement;
-        this.pointerPosition = POINTER_INITIAL_POSITION;
 
-        // If the target doesn't have a payload, then we have an error zone and we should not emit the event
-        // We should also reset the editor to IDLE
-
-        if (!target?.dataset?.payload) {
-            this.store.updateEditorState(EDITOR_STATE.IDLE);
+        const { empty, dropzone } = target.dataset;
 
+        // We can dragover the error zone
+        if (dropzone !== 'true') {
             return;
         }
 
-        const data: ClientData = JSON.parse(target.dataset.payload);
-
-        const payload = <PositionPayload>{
-            ...data
-        };
-
-        this.place.emit(payload);
-    }
-
-    /**
-     * Set the pointer position
-     *
-     * @param {DragEvent} event
-     * @memberof EmaPageDropzoneComponent
-     */
-    onDragover(event: DragEvent): void {
-        event.stopPropagation();
-        event.preventDefault();
-
-        const target = event.target as HTMLDivElement;
-
-        const parentReact = this.el.nativeElement.getBoundingClientRect();
+        this.calculatePosition(event);
+        const parentRect = this.el.nativeElement.getBoundingClientRect();
         const targetRect = target.getBoundingClientRect();
-        const isTop = this.isTop(event);
-
-        this.pointerPosition = {
-            left: `${targetRect.left - parentReact.left}px`,
-            width: `${targetRect.width}px`,
-            opacity: '1',
-            top: isTop
-                ? `${targetRect.top - parentReact.top}px`
-                : `${targetRect.top - parentReact.top + targetRect.height}px`,
-            height: '3px'
-        };
-    }
 
-    /**
-     * Set the pointer position
-     *
-     * @param {DragEvent} event
-     * @memberof EmaPageDropzoneComponent
-     */
-    onDragoverEmptyContainer(event: DragEvent): void {
-        event.stopPropagation();
-        event.preventDefault();
+        const isEmpty = empty === 'true';
 
-        const target = event.target as HTMLDivElement;
-
-        const parentReact = this.el.nativeElement.getBoundingClientRect();
-        const targetRect = target.getBoundingClientRect();
+        const opacity = isEmpty ? '0.1' : '1';
```

Comment:
Taking a look to this component with intensive calculation, What do you think of using more intensive `Signals`?

Reply:
I made an small refactor